matter_server/client/client.py

```diff
@@ -48,6 +46,10 @@
 )
 
 if TYPE_CHECKING:
+    from collections.abc import Callable
```

Comment:
I don't like this rule, that names that are only used for type checking should be imported under `if TYPE_CHECKING` flag. I think that's just annoying. I don't think there's big performance problem.

https://docs.astral.sh/ruff/rules/typing-only-standard-library-import/

Reply:
We can disable it if we want ?

Reply:
@frenck @MartinHjelmare do we want this rule to be disabled by default ?

Reply:
disabled in #567